### ISSUE TITLE
Premium Capacity and scheduled refresh memory usage

### ISSUE THREAD

**Issue by jonwbstr** [MicrosoftDocs/powerbi-docs#3922]
It would be helpful if the question **How is Power BI Pro different than Power BI Premium?** referenced that there are two types of premium, PPU and Capacity. This question talks about PPU but doesn't mention capacity. Capacity is covered further down but doesn't really address how it is different from Pro or PPU.

The question **What do you mean by "capacity"?** goes into this question briefly but doesn't address a significant difference between Power BI Pro and Power BI Premium Capacity.

When dealing with Premium Capacity the capacity of the premium workspace is dedicated to your organization. With Pro, the capacity is shared with other datasets in the region which is greater than the A1 Premium capacity provides. So, reports may fail after upgrading to Premium Capacity (not sure about PPU, probably not).

This is important because a common scenario is to develop a report in PowerBI Pro, and then attach premium capacity. When changing the workspace from Pro to Premium, and the capacity limits change, a report that refreshed successfully in the Power BI Pro workspace may start to fail. The failure happens when the Power BI Pro capacity has higher limits than the new Power BI Premium capacity.

Guidance on how to determine the right capacity for reports currently published to a Pro workspace with scheduled refreshing configured would be a welcome addition to the online material.

This article talks about capacity planning, but does not appear to address dataset refresh usage.
https://docs.microsoft.com/en-us/power-bi/developer/embedded/embedded-capacity-planning

This article talks about the capacity limits for Premium, but doesn't address the question of what capacity considerations need to be made when making the transition to Premium Capacity.
https://docs.microsoft.com/en-us/power-bi/enterprise/service-premium-what-is

I opened an issue about this on that page and the author doesn't feel this is the appropriate page to share that type of information.
https://github.com/MicrosoftDocs/powerbi-docs/issues/3871

I've found a few articles that talk about using PowerBI Desktop to analyze performance, but nothing concrete that says, this is memory  needed to refresh the query that you should plan for when choosing the right premium capacity. 

This article details the Max Memory per query of the different capacities, looking for something that helps us determine how much memory is needed to query existing datasets.
https://docs.microsoft.com/en-us/power-bi/enterprise/service-premium-what-is

I have an open ticket on the challenge we are facing and received the following update last night. Paraphrasing here

> There are two type of operations, Queries and commands. A Refresh (Command) is different from the memory limit for queries (Queries). Commands and Queries have different memory limits, so the error is actually expected.
>  
> On a Gen2 capacity, the effective memory that’s available to a command is SKU Size – Db Size. For example, a customer on an A1 SKU (3 GB) attempting to refresh (command) their dataset which is ~1.4 GB, leaves ~1.6GB of effective memory available to refresh their dataset.
>  
> Since there’s no customer capacity in Shared, this is one of the rare situations where the dataset has more memory available to it in Shared than in Premium, because they’ve chosen the smallest SKU.
>
> In this scenario the solution is to go for a larger SKU or reduce the amount of data imported into the Dataset to reduce the memory footprint of your dataset.

This may not be the right article to address these questions fully, but I wouldn't have needed to open a support ticket if this information was available somewhere. Apologies if it is and me and the engineer assisting me were unable to find it.

Hope this helps,
-jon

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8fd36f51-0d23-5363-7173-2f373239c431
* Version Independent ID: 8fd36f51-0d23-5363-7173-2f373239c431
* Content: [Power BI Premium FAQ - Power BI](https://docs.microsoft.com/en-us/power-bi/enterprise/service-premium-faq#how-is-power-bi-pro-different-than-power-bi-premium---)
* Content Source: [powerbi-docs/enterprise/service-premium-faq.yml](https://github.com/MicrosoftDocs/powerbi-docs/blob/main/powerbi-docs/enterprise/service-premium-faq.yml)
* Service: **powerbi**
* Sub-service: **powerbi-premium**
* GitHub Login: @KesemSharabi
* Microsoft Alias: **kesharab**

**Comment by jonwbstr** [comment]
>On a Gen2 capacity, the effective memory that’s available to a command is SKU Size – Size of all datasets in the workspace. For example, a customer on an A1 SKU (3 GB) attempting to refresh (command) their datasets total  is ~1.4 GB, leaves ~1.6GB of effective memory available to refresh a dataset.

**Comment by jonwbstr** [comment]
This article might be a good home for some of this content with a breadcrumb from the FAQ to the article below.

https://docs.microsoft.com/en-us/power-bi/enterprise/service-premium-capacity-optimize#why-are-data-refreshes-not-completing

Also received this additional piece of information today,
> When the model is refreshed the dataset gets uncompressed and In shared capacity, the service places a limit on the amount of uncompressed data that is processed during refresh to 10 GB. 

To recap Power BI Pro has 10GB available for a command refresh while a premium workspace is limited to SKU Size - Size of all datasets assigned to the capacity.

**Comment by jonwbstr** [comment]
I spoke to a MS Pre-Sales advisor and here is the update I received. This helped us tremendously in choosing the right SKU. We understand it's still a rough guess but it's simple and gets us in the ballpark much better than previous guidance.

TLDR: Use the memory of the msmdsrv.exe process and use the following formula Mem+mem+mem*0.7

Eg: msmdsrv.exe is using 500MB: 500+500+500*.7 = 1,350MB.

Full details below
---
While it is very difficult to predict the exact memory usage of a report in the capacity, following can be used to get an idea of the best SKU: 
1. First, find the memory footprint of the report running in the Power BI Desktop. This size would be different from the pbix file. To find the memory footprint size, please check the msmdsrv.exe entry in the Task manager. For details please refer: [Power BI Premium Performance – Part 5 | How much memory will the PBIX consume? - FourMoo | Power BI | Data Analytics](https://www.fourmoo.com/2019/06/06/power-bi-premium-performance-part-5-how-much-memory-will-the-pbix-consume/) (Third party article)
 2. Using this number, which is the size of the memory occupied by the report when it rendered, and an equivalent size is occupied for the refresh. 35% of the total value (of the dataset and the refresh) is considered for operations on the report. (Thus for example if there is one dataset, it’ll be around x + x + 0.7x ). This should give a tentative value of memory needed in the capacity.
    ![image](https://user-images.githubusercontent.com/15677312/184417864-9dd542db-615f-4af6-98d7-46262766a374.png)

This size should help you stay under the limits of a capacity SKU. Source: [Microsoft Power BI: Premium capacity - why use it how to plan for it evaluate your - BRK2046 - YouTube](https://www.youtube.com/watch?v=C6vk6wk9dcw)
 
The metrics app can also help in understanding the throttling of the  memory in the given SKU. Thus, another alternative to find the tentative SKU would be to run the metrics app alongside the report, and checking the metric for memory utilization. For metrics app, please refer: [Install the Premium Gen2 metrics app - Power BI | Microsoft Docs](https://docs.microsoft.com/en-us/power-bi/enterprise/service-premium-install-gen2-app?tabs=1st)

**Comment by KesemSharabi** [comment]
Hi @jonwbstr, 

Thank you for your very detailed feedback. I'm going to look into this with the product team and use it to improve our documentation.

Is it safe to say that the two main thinks you couldn't find, are:

- Not enough info re the transition from Pro and PPU to Premium (mainly around Pro and PPU capacity limits)
- Not having a good mechanism for calculating the impact of dataset refreshes.

Thanks,

Kesem

**Comment by jonwbstr** [comment]
Well said,

I don't mind the process being complicated, it's not that bad. Tracking the
info down was the challenge. If you can confirm what I shared is reliable,
and add it to an existing or new article my goal is met.

Thanks!
-jon

**Comment by djurecic0** [comment]
@jonwbstr I learned a lot from your posts, thank you.
